File: vertx-lang-php-jphp/src/main/php/io/vertx/php/core/Future.php
<?php 
namespace io\vertx\php\core;
class Future
{
    
    private function __construct()
    {

    }
    /**
     * ()
     * @return Throwable
     * methodSize : 1
     */
    public function cause(/*0*/)
    {

    }
    /**
     * (Memory result)
     * @return Void
     * ()
     * @return Void
     * methodSize : 2
     */
    public function complete(/*1*/$arg0)
    {

    }
    /**
     * ()
     * @return callable
     * methodSize : 1
     */
    public function completer(/*0*/)
    {

    }
    /**
     * (callable handler, Future next)
     * @return Future
     * (callable mapper)
     * @return Future
     * methodSize : 2
     */
    public function compose(/*2*/$arg0, $arg1)
    {

    }
    /**
     * (Throwable cause)
     * @return Void
     * (string failureMessage)
     * @return Void
     * methodSize : 2
     */
    public function fail(/*1*/$arg0)
    {

    }
    /**
     * ()
     * @return bool
     * methodSize : 1
     */
    public function failed(/*0*/)
    {

    }
    /**
     * (Throwable t)
     * @return Future
     * (string failureMessage)
     * @return Future
     * methodSize : 2
     */
    public function failedFuture(/*1*/$arg0)
    {

    }
    /**
     * (callable handler)
     * @return Future
     * ()
     * @return Future
     * methodSize : 2
     */
    public function future(/*1*/$arg0)
    {

    }
    /**
     * ()
     * @return bool
     * methodSize : 1
     */
    public function isComplete(/*0*/)
    {

    }
    /**
     * (callable mapper)
     * @return Future
     * (Memory value)
     * @return Future
     * methodSize : 2
     */
    public function map(/*1*/$arg0)
    {

    }
    /**
     * ()
     * @return Future
     * methodSize : 1
     */
    public function mapEmpty(/*0*/)
    {

    }
    /**
     * (callable mapper)
     * @return Future
     * (Memory value)
     * @return Future
     * methodSize : 2
     */
    public function otherwise(/*1*/$arg0)
    {

    }
    /**
     * ()
     * @return Future
     * methodSize : 1
     */
    public function otherwiseEmpty(/*0*/)
    {

    }
    /**
     * (callable mapper)
     * @return Future
     * methodSize : 1
     */
    public function recover(/*1*/$arg0)
    {

    }
    /**
     * ()
     * @return Memory
     * methodSize : 1
     */
    public function result(/*0*/)
    {

    }
    /**
     * (callable handler)
     * @return Future
     * methodSize : 1
     */
    public function setHandler(/*1*/$arg0)
    {

    }
    /**
     * ()
     * @return bool
     * methodSize : 1
     */
    public function succeeded(/*0*/)
    {

    }
    /**
     * ()
     * @return Future
     * (Memory result)
     * @return Future
     * methodSize : 2
     */
    public function succeededFuture(/*1*/$arg0)
    {

    }
    /**
     * (Memory result)
     * @return bool
     * ()
     * @return bool
     * methodSize : 2
     */
    public function tryComplete(/*1*/$arg0)
    {

    }
    /**
     * (Throwable cause)
     * @return bool
     * (string failureMessage)
     * @return bool
     * methodSize : 2
     */
    public function tryFail(/*1*/$arg0)
    {

    }
}

File: vertx-lang-php-jphp/src/main/api/io/vertx/php/core/Future.java
package io.vertx.php.core;
import php.runtime.annotation.Reflection.Name;
import php.runtime.annotation.Reflection.Namespace;
import php.runtime.annotation.Reflection.Signature;
import io.vertx.lang.jphp.IWrapper;
@Name("Future")
@Namespace("io\\vertx\\php\\core")
public interface Future<S extends io.vertx.core.Future<T>, T> extends IWrapper<S>{
    
    @Signature
    static Memory future(Environment __env__, Memory... args) {
        return Utils.convReturnVertxGen(__env__, FutureImpl::new, io.vertx.core.Future.future()
    }

    @Signature
    static Memory future(Environment __env__, Memory... args) {
        return Utils.convReturnVertxGen(__env__, FutureImpl::new, io.vertx.core.Future.future()
    }
    
    @Signature
    static Memory succeededFuture(Environment __env__, Memory... args) {
        return Utils.convReturnVertxGen(__env__, FutureImpl::new, io.vertx.core.Future.succeededFuture()
    }

    @Signature
    static Memory succeededFuture(Environment __env__, Memory... args) {
        return Utils.convReturnVertxGen(__env__, FutureImpl::new, io.vertx.core.Future.succeededFuture()
    }

    @Signature
    static Memory failedFuture(Environment __env__, Memory... args) {
        return Utils.convReturnVertxGen(__env__, FutureImpl::new, io.vertx.core.Future.failedFuture(t)
    }
    
    @Signature
    default Memory isComplete(Environment __env__, Memory... args) {
        return Utils.convReturnUnknow(__env__, this.getWrappedObject().isComplete()
    }

    @Signature
    default Memory setHandler(Environment __env__, Memory... args) {
        this.getWrappedObject().setHandler(handler)
    }
    
    @Signature
    default void complete(Environment __env__, Memory... args) {
        this.getWrappedObject().complete()
    }

    @Signature
    default void complete(Environment __env__, Memory... args) {
        this.getWrappedObject().complete()
    }

    @Signature
    default void fail(Environment __env__, Memory... args) {
        this.getWrappedObject().fail(cause)
    }
    
    @Signature
    default Memory tryComplete(Environment __env__, Memory... args) {
        return Utils.convReturnUnknow(__env__, this.getWrappedObject().tryComplete()
    }

    @Signature
    default Memory tryComplete(Environment __env__, Memory... args) {
        return Utils.convReturnUnknow(__env__, this.getWrappedObject().tryComplete()
    }

    @Signature
    default Memory tryFail(Environment __env__, Memory... args) {
        return Utils.convReturnUnknow(__env__, this.getWrappedObject().tryFail(cause)
    }
    
    @Signature
    default Memory result(Environment __env__, Memory... args) {
        return Utils.convReturnUnknow(__env__, this.getWrappedObject().result()
    }
    
    @Signature
    default Memory cause(Environment __env__, Memory... args) {
        return Utils.convReturnUnknow(__env__, this.getWrappedObject().cause()
    }
    
    @Signature
    default Memory succeeded(Environment __env__, Memory... args) {
        return Utils.convReturnUnknow(__env__, this.getWrappedObject().succeeded()
    }
    
    @Signature
    default Memory failed(Environment __env__, Memory... args) {
        return Utils.convReturnUnknow(__env__, this.getWrappedObject().failed()
    }

    @Signature
    default Memory compose(Environment __env__, Memory... args) {
        return Utils.convReturnVertxGen(__env__, FutureImpl::new, this.getWrappedObject().compose(mapper)
    }

    @Signature
    default Memory map(Environment __env__, Memory... args) {
        return Utils.convReturnVertxGen(__env__, FutureImpl::new, this.getWrappedObject().map(mapper)
    }
    
    @Signature
    default Memory mapEmpty(Environment __env__, Memory... args) {
        return Utils.convReturnVertxGen(__env__, FutureImpl::new, this.getWrappedObject().mapEmpty()
    }
    
    @Signature
    default Memory completer(Environment __env__, Memory... args) {
        return Utils.convReturnUnknow(__env__, this.getWrappedObject().completer()
    }

    @Signature
    default Memory recover(Environment __env__, Memory... args) {
        return Utils.convReturnVertxGen(__env__, FutureImpl::new, this.getWrappedObject().recover(mapper)
    }

    @Signature
    default Memory otherwise(Environment __env__, Memory... args) {
        return Utils.convReturnVertxGen(__env__, FutureImpl::new, this.getWrappedObject().otherwise(mapper)
    }
    
    @Signature
    default Memory otherwiseEmpty(Environment __env__, Memory... args) {
        return Utils.convReturnVertxGen(__env__, FutureImpl::new, this.getWrappedObject().otherwiseEmpty()
    }

}
